### PR TITLE
Application add version flag

### PR DESCRIPTION
**Pull request by ss89**

### CODE REVIEW

File: pkg/application/options.go [hunk 1]
```diff
@@ -305,3 +319,18 @@ func WithUTCClock(useUTC bool) Option {
 		})
 	}
 }
+
+func WithVersion(version string) Option {
+	return func(app *App) {
+		app.addSetupOption(func(config cfg.GosoConf, logger mon.GosoLog) error {
```
Comment (j4k4) [inline pkg/application/options.go:325]: feels weird to use a setup option to log something..

File: pkg/application/options.go [hunk 2]
```diff
@@ -262,6 +262,20 @@ func WithLoggerTagsFromConfig(app *App) {
 	})
 }
 
+func WithLoggerVersionTag(version string) Option {
```
Comment (j4k4) [inline pkg/application/options.go:265]: where is this requirement coming from?

Comment (ss89) [reply pkg/application/options.go:265]: from me :-) private project where the user of the app should somehow be able to know whether he is using an up-to-date version without the app having to call an api endpoint to determine that.

File: pkg/application/options.go (hunk 1 above)
Comment (ss89) [reply pkg/application/options.go:325]: Please #BeDirect and let me know where you would put it

Comment (j4k4) [reply pkg/application/options.go:325]: well, i don't have always a better idea right away, but my first impulse would be something like a `-version` flag for your command..

**Comment by applike-ss** [comment]
closing this, as it will never get merged and go 1.18 does now include versioning information in go builds unless specified otherwise.

**Pull request closed without merging**